k8s/pkg/apis/longhorn/v1beta1/node.go

```diff
@@ -32,53 +32,90 @@ const (
 )
 
 type DiskSpec struct {
-	Path              string   `json:"path"`
-	AllowScheduling   bool     `json:"allowScheduling"`
-	EvictionRequested bool     `json:"evictionRequested"`
-	StorageReserved   int64    `json:"storageReserved"`
-	Tags              []string `json:"tags"`
+	// +optional
+	Path string `json:"path"`
+	// +optional
+	AllowScheduling bool `json:"allowScheduling"`
+	// +optional
+	EvictionRequested bool `json:"evictionRequested"`
+	// +optional
+	StorageReserved int64 `json:"storageReserved"`
+	// +optional
+	Tags []string `json:"tags"`
 }
 
 type DiskStatus struct {
-	Conditions       map[string]Condition `json:"conditions"`
-	StorageAvailable int64                `json:"storageAvailable"`
-	StorageScheduled int64                `json:"storageScheduled"`
-	StorageMaximum   int64                `json:"storageMaximum"`
-	ScheduledReplica map[string]int64     `json:"scheduledReplica"`
-	DiskUUID         string               `json:"diskUUID"`
+	// +optional
+	Conditions map[string]Condition `json:"conditions"`
+	// +optional
+	StorageAvailable int64 `json:"storageAvailable"`
+	// +optional
+	StorageScheduled int64 `json:"storageScheduled"`
+	// +optional
+	StorageMaximum int64 `json:"storageMaximum"`
+	// +optional
+	ScheduledReplica map[string]int64 `json:"scheduledReplica"`
+	// +optional
+	DiskUUID string `json:"diskUUID"`
 }
 
+// NodeSpec defines the desired state of the Longhorn node
 type NodeSpec struct {
-	Name                     string              `json:"name"`
-	Disks                    map[string]DiskSpec `json:"disks"`
-	AllowScheduling          bool                `json:"allowScheduling"`
-	EvictionRequested        bool                `json:"evictionRequested"`
-	Tags                     []string            `json:"tags"`
-	EngineManagerCPURequest  int                 `json:"engineManagerCPURequest"`
-	ReplicaManagerCPURequest int                 `json:"replicaManagerCPURequest"`
+	// +optional
+	Name string `json:"name"`
+	// +optional
+	Disks map[string]DiskSpec `json:"disks"`
+	// +optional
+	AllowScheduling bool `json:"allowScheduling"`
+	// +optional
+	EvictionRequested bool `json:"evictionRequested"`
+	// +optional
+	Tags []string `json:"tags"`
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=40
```

Comment:
Why is the EngineManagerCPURequest's Maximum 40?

Reply:
I check the doc and the chart questions.yaml, the max value is 40.

Reply:
After the system upgrade, there may be 2 sets of instance manager pods on each node. To avoid the instance manager pods eating up all CPU resources, we just limit the max number to 40%.